### PR TITLE
Hide health tabs for disenrolled patients from non-health-admin users

### PR DESCRIPTION
**Pull request by gigxz**
- [x] Comment out all `set_patient if @patient.blank?` usages
- [x] Don't show health tabs on the dashboard if the current user cannot access the patient (because they are disenrolled)
- [x] Redirect to 401 page from health controllers if user cannot access patient

### CODE REVIEW

File: app/models/health/patient.rb
Comment (gigxz) [inline app/models/health/patient.rb:344]: @awisema we have several scope and functions now for "visibility" of patients..
1) This one, `viewable_by_user` which limits by `not_confirmed_rejected` referrals. This is used for the client dashboard.


2) `def self.accessible_by_user user` which limits by `can_edit_client_health` and `can_view_client_health` permissions, as well as `consent_revoked`. This is only used by the Pilot  controllers as far as I can see

3) `def visible_to(user)` is similar to `accessible_by_user` but also limits by health_agency. It's used [here](https://github.com/greenriver/hmis-warehouse/blob/acba4c22036465e2649e64c0c7e17b0a00cee24f/app/controllers/clients_controller.rb#L124)

**Review by awisema — APPROVED** [review]
This looks good, thank you! I'll wait until the tests finish, and then merge.

**Pull request merged**
